SpokestackTray/Controllers/SpeechController.swift

```diff
@@ -0,0 +1,492 @@
+                    
+                    return metaDataURL
+                }
+                .eraseToAnyPublisher()
+    }
+
+    private var downloadNLUMetaData: AnyPublisher<URL, Error> {
+        
+        let url: URL = URL(string: configuration.nluModelURLs[NLUModelURLMetaDataKey]!)!
+        return URLSession.shared.dataTaskPublisher(for: url)
+                .tryMap { response -> URL in
+                        
+                    guard let httpURLResponse = response.response as? HTTPURLResponse, httpURLResponse.statusCode == 200 else {
+                        throw TraySetupInitializeError.invalidModelDownloadStatus
+                    }
+                    
+                    let metaDataURL: URL = URL.spsk_documentsDirectory.appendingPathComponent(url.lastPathComponent)
+                    try? response.data.write(to: metaDataURL)
+                    
+                    return metaDataURL
+                }
+                .eraseToAnyPublisher()
+    }
+    
+    private var downloadNLU: AnyPublisher<URL, Error> {
+        
+        let url: URL = URL(string: configuration.nluModelURLs[NLUModelURLNLUKey]!)!
+        return URLSession.shared.dataTaskPublisher(for: url)
+                .tryMap { response -> URL in
+                        
+                    guard let httpURLResponse = response.response as? HTTPURLResponse, httpURLResponse.statusCode == 200 else {
+                        throw TraySetupInitializeError.invalidModelDownloadStatus
+                    }
+                    
+                    let metaDataURL: URL = URL.spsk_documentsDirectory.appendingPathComponent(url.lastPathComponent)
+                    try? response.data.write(to: metaDataURL)
+                    
+                    return metaDataURL
+                }
+                .eraseToAnyPublisher()
+    }
+    
+    private var downloadNLUVocab: AnyPublisher<URL, Error> {
+        
+        let url: URL = URL(string: configuration.nluModelURLs[NLUModelURLVocabKey]!)!
+        return URLSession.shared.dataTaskPublisher(for: url)
+                .tryMap { response -> URL in
+                        
+                    guard let httpURLResponse = response.response as? HTTPURLResponse, httpURLResponse.statusCode == 200 else {
+                        throw TraySetupInitializeError.invalidModelDownloadStatus
+                    }
+                    
+                    let metaDataURL: URL = URL.spsk_documentsDirectory.appendingPathComponent(url.lastPathComponent)
+                    try? response.data.write(to: metaDataURL)
+                    
+                    return metaDataURL
+                }
+                .eraseToAnyPublisher()
+    }
+
+    lazy private var pipeline: SpeechPipeline = {
+
+         return try! SpeechPipelineBuilder()
+                    .addListener(self)
+                    .setDelegateDispatchQueue(DispatchQueue.main)
+                    .useProfile(.tfLiteWakewordAppleSpeech)
```

Comment:
more of a future note than a PR change: tray should support the client specifying the profile (which has implications for how the rest of the initialization is handled), but I think that will be an additive change to tray in the future.

Reply:
I went ahead and added the configuration value and default to `. tfLiteWakewordAppleSpeech`